### PR TITLE
feat: Cluster -> Environment - namespace labels support

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/devtron-labs/devtron/issues/5715

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/utils.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+              namespace: data.namespace || '',
+              active: true,
+              default: data.isProduction,
+              description: data.description || '',
+              updateLabels: !!namespaceLabels,
+              ...(namespaceLabels
+                  ? {
+                        namespaceResourceVersion: resourceVersion,
+                        namespaceLabels: namespaceLabels
+                            .filter(({ key, value }) => !!key && !!value)
```
Comment (eshankvaish) [inline src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/utils.ts:36]:
```suggestion
                            .filter(({ key, value }) => !!key.trim() && !!value.trim())
```

File: src/components/cluster/Cluster.tsx [hunk 2]
```diff
@@ -864,18 +851,13 @@ const Cluster = ({
...
             </article>
             {showWindow && (
-                <Drawer position="right" width="800px" onEscape={hideClusterDrawer}>
-                    <div className="h-100 bcn-0" ref={editLabelRef}>
-                        <Environment
-                            reload={reload}
-                            cluster_name={cluster_name}
-                            {...environment}
-                            hideClusterDrawer={hideClusterDrawer}
-                            isNamespaceMandatory={!isVirtualCluster}
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/components/cluster/Cluster.tsx:874]: discussed this will be changed on BE

**Review by eshankvaish — DISMISSED** [review]

File: src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/types.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    clusterId: number
+    prometheusEndpoint: string
+    isProduction: boolean
+    description: string
+    reload: () => void
+    hideClusterDrawer: () => void
+    isVirtual: boolean
+}
+
+export interface ClusterEnvironmentDrawerFormProps {
```
Comment (vivek-devtron) [inline src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/types.ts:16]: We should extend `ClusterEnvironmentDrawerProps` from `ClusterEnvironmentDrawerFormProps`

File: src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/ClusterEnvironmentDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+        hideClusterDrawer()
+        reload()
+    }
+
+    const setTags = (tags: TagType[]) => setNamespaceLabels((prev) => ({ ...prev, labels: tags }))
+
+    const addLabel = async () => {
+        const nameSpaceErr = trigger('namespace')
+        if (!nameSpaceErr) {
+            await fetchClusterNamespaces(data.namespace)
```
Comment (vivek-devtron) [inline src/Pages/GlobalConfigurations/ClustersAndEnvironments/ClusterEnvironmentDrawer/ClusterEnvironmentDrawer.tsx:217]: We should call `refetchNamespaceLabels`

**Review by vivek-devtron — APPROVED** [review]

File: src/stories/useForm.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+const FormComponent = ({ validationMode }: { validationMode: 'onChange' | 'onBlur' }) => {
+    const { data, errors, register, handleSubmit, formState } = useForm<FormData>({
+        initialValues: { email: '', password: '' },
+        validations,
+        validationMode,
+    })
+
+    const onSubmit: UseFormSubmitHandler<FormData> = (data) => {
+        alert(JSON.stringify(data, null, 2))
```
Comment (eshankvaish) [inline src/stories/useForm.stories.tsx:44]: use action instead

**Review by eshankvaish — DISMISSED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**